src/style_manager/model/PropertyFactory.js

```diff
@@ -270,6 +300,29 @@ module.exports = () => ({
           break;
       }
 
+      /*
+       * Add styleable dependency on other properties. Allows properties to be
+       * dynamically hidden or shown based on values of other properties.
+       *
+       * Property will be styleable if all of the properties (keys) in the
+       * requires object have any of the values specified in the array.
+       */
+      switch (prop) {
+        case 'flex-direction':
+        case 'flex-wrap':
+        case 'justify-content':
+        case 'align-items':
+        case 'align-content':
+          obj.requires = { display: ['flex'] };
+          break;
+        case 'order':
+        case 'flex-grow':
+        case 'flex-shrink':
+        case 'align-self':
+          obj.requiresParent = { display: ['flex'] };
+          break;
+      }
+
```

Comment:
It would be great if you add those properties to `defaults` (in style_manager/model/Property.js) and write a tiny comment about the usage, otherwise, we risk to forget about them